interfaces/devdoc/timer_requirements.md

```diff
@@ -0,0 +1,96 @@
+# timer
+================
+
+## Overview
+
+`timer` provides platform-independent time measurement primitives.
+
+## Exposed API
+
+```c
+MOCKABLE_FUNCTION(, TIMER_HANDLE, timer_create_new);
+MOCKABLE_FUNCTION(, int, timer_start, TIMER_HANDLE, handle);
+MOCKABLE_FUNCTION(, double, timer_get_elapsed, TIMER_HANDLE, timer);
+MOCKABLE_FUNCTION(, double, timer_get_elapsed_ms, TIMER_HANDLE, timer);
+MOCKABLE_FUNCTION(, void, timer_destroy, TIMER_HANDLE, timer);
+MOCKABLE_FUNCTION(, double, timer_global_get_elapsed_ms);
+MOCKABLE_FUNCTION(, double, timer_global_get_elapsed_us);
+```
+
+### timer_create_new
+
+```c
+MOCKABLE_FUNCTION(, TIMER_HANDLE, timer_create_new);
+```
+
+`timer_create_new` creates a new timer.
+
+**SRS_TIMER_01_001: [** `timer_create_new` shall create a new timer and on success return a non-`NULL` handle to it. **]**
+
+### timer_destroy
+
+```c
+MOCKABLE_FUNCTION(, void, timer_destroy, TIMER_HANDLE, timer);
+```
+
+`timer_destroy` frees all resources associated with `timer`.
+
+**SRS_TIMER_01_002: [** If `timer` is `NULL`, `timer_destroy` shall return. **]**
+
+**SRS_TIMER_01_003: [** Otherwise, `timer_destroy` shall free the memory associated with `timer`. **]**
+
+### timer_start
+
+```c
+MOCKABLE_FUNCTION(, int, timer_start, TIMER_HANDLE, timer);
+```
+
+`timer_start` starts the timer.
+
+**SRS_TIMER_01_004: [** If `timer` is `NULL`, `timer_start` shall fail and return a non-zero value. **]**
+
+**SRS_TIMER_01_005: [** Otherwise, `timer_start` shall record the start time (used for computing the elapsed time). **]**
+
+### timer_get_elapsed
+
+```c
+MOCKABLE_FUNCTION(, double, timer_get_elapsed, TIMER_HANDLE, timer);
+```
+
+`timer_get_elapsed` returns the time in seconds elapsed since the timer was started.
+
+**SRS_TIMER_01_006: [** If `timer` is `NULL`, `timer_get_elapsed` shall return -1. **]**
```

Comment:
> -1 [](start = 78, length = 3)

requirement 01_008 says "a negative value". I think they should be consistent either way (-1 or a negative value)

Reply:
Changed to -1.

---
In reply to: [512802673](https://github.com/Azure/azure-c-pal/pull/48#discussion_r512802673) [](ancestors = 512802673)